modules/viBidAdapter.js

```diff
@@ -0,0 +1,69 @@
+import * as utils from 'src/utils';
+import { registerBidder } from 'src/adapters/bidderFactory';
+import { BANNER } from 'src/mediaTypes';
+
+const BIDDER_CODE = 'vi';
+const SUPPORTED_MEDIA_TYPES = [BANNER];
+
+function isBidRequestValid(bid) {
+  return !!(bid.params.pubId);
+}
+
+function buildRequests(bidReqs) {
+  let imps = [];
+  utils._each(bidReqs, function (bid) {
+    imps.push({
+      id: bid.bidId,
+      sizes: bid.sizes,
```

Comment:
ad unit sizes may be defined as a double array (`[[320, 480]]`) or single array (`[320, 480]`). In the case of single sizes, your endpoint is erroring out with `400 (Invalid JSON: Cannot deserialize instance of int[] out of VALUE_NUMBER_INT`, but `utils.parseSizesInput` can help. Something like
```
utils.parseSizesInput(bid.sizes).map(size => size.split('x'))
```
though feel free handle it however you'd like